augur/metrics/commit.py

```diff
@@ -111,60 +115,64 @@ def annual_commit_count_ranked_by_new_repo_in_repo_group(self, repo_group_id, re
     if not end_date:
         end_date = datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')
 
-    
+
 
     cdRgNewrepRankedCommitsSQL = None
 
     if not repo_id:
         table = 'dm_repo_group_annual' if period == 'year' or period == 'all' else 'dm_repo_group_monthly' if period == 'month' else 'dm_repo_group_weekly'
-        cdRgNewrepRankedCommitsSQL = s.sql.text("""
-            SELECT repo_groups.repo_group_id, rg_name, year, sum(cast(added AS INTEGER) - cast(removed AS INTEGER) - cast(whitespace AS INTEGER)) AS net, sum(cast(patches AS INTEGER)) AS commits
-            FROM {0}, repo_groups
-            WHERE {0}.repo_group_id = repo_groups.repo_group_id            
-            AND repo_groups.repo_group_id = :repo_group_id
-            AND (
-                year > date_part('year', TIMESTAMP :begin_date)
-                OR (
-                    year = date_part('year', TIMESTAMP :begin_date) 
-                    AND {1} >= date_part('{1}', TIMESTAMP :begin_date)
-                )
-            )
-            AND (
-                year < date_part('year', TIMESTAMP :end_date)
-                OR (
-                    year = date_part('year', TIMESTAMP :end_date) 
-                    AND {1} <= date_part('{1}', TIMESTAMP :end_date)
-                )
-            )
-            GROUP BY repo_groups.repo_group_id, rg_name, YEAR
-            ORDER BY YEAR ASC
-        """.format(table, period))
+        cdRgNewrepRankedCommitsSQL = SELECT([TEXT('repo_groups.repo_group_id, rg_name, year, sum(cast(added AS INTEGER) - cast(removed AS INTEGER) - cast(whitespace AS INTEGER)) AS net, sum(cast(patches AS INTEGER)) AS commits')]).\
```

Comment:
Please keep the raw SQL, and just change the format of the parameter binding. We use raw SQL statements for every other metric, and having the raw SQL be easily readable is important especially for the metrics in order to understand exactly how they are built, and the format of the data they return.

Reply:
@ccarterlandis : by "raw sql" do you mean "formatted sql"?  I think this is just formatted poorly, but it is "raw sql". Am I right?

Reply:
@sgoggins yes, that's what I mean. I chose my words poorly. Besides wanting to remain consistent in our implementation of the metrics, I personally don't like using the `.where` and `.and` methods. AFAIK, they provide no functional benefits over encompassing everything in a `.text` , and they only obscure the actual structure of the query. In our case specifically, we need the metrics to be as easily readable as possible, which is why I requested the change